doc/prscheduler.md

```diff
@@ -0,0 +1,157 @@
+credentials:
+  token: $github_token
+```
+
+
+### Ansible playbook
+
+#### Use the playbook
+
+In order to install and configure the scheduler tool on a machine, you can
+use the ansible playbook `ansible/prepare_openclose_pr_tool.yml`. You need
+first to edit the inventory file `ansible/hosts/runners` and define the
+machine that will be used as scheduler (the playbook will apply to all
+ the machines in the [scheduler] group):
+```
+# cat ansible/hosts/runners
+[scheduler]
+1.2.3.4        # the scheduler IP
+```
+
+Then run:
+```
+ansible-playbook -i ansible/hosts/runners ansible/prepare_openclose_pr_tool.yml
+```
+
+The playbook will prompt you for variables:
+```
+# ansible-playbook -i ansible/hosts/runners ansible/prepare_openclose_pr_tool.yml
+Provide an unique identifier for the PR: testing_master    # $identifier
+Branch name to open PR against it: master                  # $branch
+Repo owner to create branches: freeipa-pr-ci               # $repo_owner
+GitHub token: 123qwerty                                    # $github_token
+Provide the path to a test definition file to use it to open new PRs. For example: ipatests/prci_definition/gating	# $prci_config, for instance ipatests/prci_definition/nightly_master.yaml
+Provide a ssh key that has push access to the git repo: ~/path/to/ssh_key #$git_sshkey
+Should the PR be open against the upstream repo?. Type no for opening it agaist your own freeipa repo (yes/no): no
+```
+
+You can run the ansible playbook multiple times to schedule multiple PRs,
+but to each PR you want to create you need to provide a different
+identifier (ID). This ID is used to create the branch and to name the
+PR.
+
+
+#### Playbook tasks
+
+The playbook is performing multiple tasks in order to install and configure
+the python script:
+* installation of python and deps for ansible modules (python2, python2-dnf,
+libselinux-python)
+* prompting of required variables (identifier, branch, repo_owner,
+github_token, prci_config, git_sshkey, pr_against_upstream)
+* installation of the role `automation/openclose_pr`:
+  * installation of required packages: python3-github3py, PyYAML, git
+  * installation of python dependencies: CacheControl, gitpython
+  * creation of the base repository where the tool and config is stored:
+`/root/openclose_pr`
+  * creation of the configuration file `/root/openclose_pr/config.yaml`
+containing repo info and credentials
+  * installation of the python script in `/root/openclose_pr/openclose_pr.py`
+  * linking of the test definition file 
+`/root/openclose_pr/.freeipa-pr-ci.yaml` to $prci_config
+  * preparation of ssh configuration in order to access github.com
+  * creation of a clone repo from $repo_owner/freeipa.git to
+`/root/openclose_pr/freeipa/`
+  * configuration of the cron job to run daily at 23:00
+
+#### Current scheduling
+
+Currently we are using a machine called prci-automation in our OpenStack lab,
+with cron job starting at 23:45 every monday, wednesday and friday,
```

Comment:
I think we should decide on what to hard-code here. Either everyday or mo, we, fri. 3x per week is probably ok.

Reply:
I agree it would be better to align the code in freeipa-pr-ci and the lab config. We should use mon, we, fri in the lab, meaning that I need to remove that comment and modify the crontab in the  task /ansible/roles/automation/openclose_pr/tasks/main.yml to be consistent with this decision.